pandas/io/packers.py

```diff
@@ -371,9 +385,8 @@ def encode(obj):
             return {'typ': 'block_manager',
                     'klass': obj.__class__.__name__,
                     'axes': data.axes,
-                    'blocks': [{'items': data.items.take(b.mgr_locs),
-                                'locs': b.mgr_locs.as_array,
-                                'values': convert(b.values),
+                    'blocks': [{'locs': b.mgr_locs.as_array,
+                                'values': convert_if_not_dt_index(b.values),
```

Comment:
if it gets converted to a numpy array now, wouldn't the whole `encode`/`decode` for `DatetimeIndex` be skipped?  Or is there a way to construct DatetimeTZBlock from numpy arrays?